runtime-sdk/modules/evm/src/evm_backend.rs

```diff
@@ -166,82 +148,45 @@ impl<'c, C: oasis_runtime_sdk::Context> ApplyBackend for Backend<'c, C> {
+                    if let Some(code) = code {
                         let state = self.ctx.get_mut().runtime_state();
-                        let a_store = storage::PrefixStore::new(state, &crate::MODULE_NAME);
-                        let mut accounts = storage::TypedStore::new(storage::PrefixStore::new(
-                            a_store,
-                            &state::ACCOUNTS,
-                        ));
-                        let mut account: Account = accounts.get(&addr).unwrap_or_default();
-
-                        account.balance = basic.balance.into();
-                        account.nonce = basic.nonce.into();
-
-                        accounts.insert(&addr, account.clone());
-
-                        if let Some(code) = code {
-                            let state = self.ctx.get_mut().runtime_state();
-                            let c_store = storage::PrefixStore::new(state, &crate::MODULE_NAME);
-                            let mut codes = storage::TypedStore::new(storage::PrefixStore::new(
-                                c_store,
-                                &state::CODES,
-                            ));
-                            codes.insert(&addr, code);
-                        }
+                        let mut store = state::codes(state);
+                        store.insert(&addr, code);
+                    }
 
-                        if reset_storage {
-                            let state = self.ctx.get_mut().runtime_state();
-                            let s_store = storage::PrefixStore::new(state, &crate::MODULE_NAME);
-                            let mut storages = storage::PrefixStore::new(s_store, &state::STORAGES);
-                            storages.remove(addr.as_bytes());
-                        }
+                    // Handle storage reset.
+                    if reset_storage {
+                        // NOTE: Storage cannot be efficiently reset as this would require iterating
+                        //       over all of the storage keys. We could add this if remove_prefix
+                        //       existed.
+                    }
 
-                        for (index, value) in storage {
-                            let idx: H256 = index.into();
-                            let val: H256 = value.into();
+                    // Handle storage updates.
+                    for (index, value) in storage {
+                        let idx: H256 = index.into();
+                        let val: H256 = value.into();
 
-                            let state = self.ctx.get_mut().runtime_state();
-                            let s_store = storage::PrefixStore::new(state, &crate::MODULE_NAME);
-                            let storages = storage::PrefixStore::new(s_store, &state::STORAGES);
-                            let mut s = storage::TypedStore::new(storage::HashedStore::<
-                                _,
-                                blake3::Hasher,
```

Comment:
what was this about a hasher in the previous implementation? is that removed in this pr? if so, what will that do?

Reply:
It is not removed, just refactored so that it is not repeated over and over. See the `state::storage` function.

Reply:
found it, thanks